### PR TITLE
[PICA-194] Update extend functionality

### PR DESCRIPTION
**Pull request by benluelo**

### CODE REVIEW

**Comment by KaiserKarel** [comment]
Task linked: [PICA-194 Extending staking position time design and implementation](https://app.clickup.com/t/20465559/PICA-194)

File: code/parachain/frame/composable-traits/src/staking/mod.rs [hunk 1]
```diff
@@ -317,3 +324,13 @@ pub trait Staking {
 	/// * `position` - The uniquely identifying NFT from which we will compute the rewards.
 	fn claim(who: &Self::AccountId, position: &Self::PositionId) -> DispatchResult;
 }
+
+/// How much to extend a stake's duration by, used by [`Staking::extend_stake_duration`].
+#[derive(RuntimeDebug, PartialEq, Eq, Clone, Encode, Decode, TypeInfo)]
+pub enum StakeDurationExtendAmount {
```
Comment (vimukthi-git) [inline code/parachain/frame/composable-traits/src/staking/mod.rs:330]:
```suggestion
pub enum StakeDurationExtention {
```

File: code/parachain/frame/composable-traits/src/staking/lock.rs [hunk 2]
```diff
@@ -60,19 +62,48 @@ impl<MaxDurationPresets: Get<u32>> DurationMultipliers<MaxDurationPresets> {
...
 	}
+
+	/// Get the maximum stake duration allowed by this duration configuration. Returns [`None`] if
+	/// the stake durations are unbounded.
+	// TODO(benluelo): We need to verify that the duration presets are valid (i.e at least one
+	// duration provided)
+	pub fn max_duration(&self) -> Option<u64> {
+		match self {
+			DurationMultipliers::Presets(presets) =>
+				presets.last_key_value().map(|(k, _)| k).copied(),
```
Comment (vimukthi-git) [inline code/parachain/frame/composable-traits/src/staking/lock.rs:73]: So we asssume last key is the max?

**Review by vimukthi-git — APPROVED** [review]
LGTM

File: code/parachain/frame/composable-traits/src/staking/lock.rs (hunk 2 above)
Comment (benluelo) [reply code/parachain/frame/composable-traits/src/staking/lock.rs:73]: Yeah, that's how BTreeMaps work: https://doc.rust-lang.org/stable/std/collections/struct.BTreeMap.html#method.last_key_value

**Comment by josietyleung** [comment]
killing as killing fnft

**Pull request closed without merging**